src/main/java/com/github/krzysiekjodlowski/ox/UI.java

```diff
@@ -0,0 +1,25 @@
+package com.github.krzysiekjodlowski.ox;
+
+/**
+ * Interacting directly with user,
+ * used for collecting input end
+ * outputting messages to user
+ *
+ * @author Krzysztof Jodlowski
+ */
+interface UI<T1, T2 extends Number> {
+    /**
+     * Outputs message to user.
+     *
+     * @param message any appropriate representation
+     */
+    void say(T1 message);
```

Comment:
Think about object as argument

Reply:
Actually behind the scenes it is an object, but parametrized one. Not sure right now but it is possible that generic nature of this interface will be abandoned.